Source/SuperLinq/FallbackIfEmpty.cs

```diff
@@ -40,16 +40,21 @@ public static IEnumerable<T> FallbackIfEmpty<T>(this IEnumerable<T> source, IEnu
 		Guard.IsNotNull(source);
 		Guard.IsNotNull(fallback);
 
-		return Core(source, fallback);
+		return source.TryGetCollectionCount() is not null
```

Comment:
`Count` as a property should not require enumeration to run. As such, we need to enforce that both `source` and `fallback` are valid before passing to the collection iterator: 
```suggestion
		return source.TryGetCollectionCount() is not null 
			&& fallback.TryGetCollectionCount() is not null
```